### PR TITLE
Add Docker and Kubernetes samples

### PR DESCRIPTION
**Pull request by marrobi**
## Description
This PR contains a Dockerfile, and examples of how to run the FHIR Server using Docker Compose, or on Kubernetes for Dev/Test scenarios.

Upstream PR - https://github.com/microsoft/fhir-server/pull/754 - wanted to check builds pass.

### CODE REVIEW

**Comment by bengreenier** [comment]
Do environment variables work for passwords/conn strings? Seems ok to me, not sure if k8s has some better option.

env vars should work just fine - Per the [Environments docs](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/environments?view=aspnetcore-3.0). Further, it's already used for the development environment - see SqlServer param overrides from environment variables [here](https://github.com/marrobi/fhir-server/blob/master/src/Microsoft.Health.Fhir.R4.Web/Properties/launchSettings.json#L16). 

Probably the most reliable and easiest thing to do is build the entire solution using `dotnet` rather than scoping to any particular projects. I __think__ it should notice the sln and do this by default (so maybe just remove your flag for Stu3).

Separating build and restore makes sense to me - potentially saves us time if restore is cachable by docker. Just be sure to do `--no-restore` on the build/publish, ofc.

**Comment by marrobi** [comment]
Yup, so I tried all sorts. So there are colons - which aren't allowed in docker, so I tried replacing `:` with `__` but they didn't get picked up. 

I had: `ASPNETCORE_SqlServer__ConnectionString`

**Comment by bengreenier** [comment]
😭 Of course not. That would be Too Easy ™

Why've you prefixed with ASPNETCORE there? Naively, it seems to be it should just be `SqlServer__ConnectionString`..

**Comment by marrobi** [comment]
Re building the whole thing - was just trying to speed up the build - it includes a sample apps, tools and cosmos stuff - which if using a SQL container wouldn't be needed.

The Stu3 flag is just for publish - I'm building the whole thing - or am I misunderstanding `dotnet` build/publish.

**Comment by bengreenier** [comment]
Ah no you're right - didn't catch the separate build command - publish will build/publish if no build is present, so thought you were relying on that.

**Comment by marrobi** [comment]
Let me try without the prefix. I might have not done that with the `__`

**Comment by marrobi** [comment]
@bengreenier got the env vars working - thanks! thought I had tried everything.

File: samples/kubernetes/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Running in Kubernetes
```
Comment (jjcollinge) [inline samples/kubernetes/README.md]: This is running in AKS not agnostic Kubernetes - sample should be explicit about that or include just the K8 commands

Comment (marrobi) [reply samples/kubernetes/README.md]: Thanks, good point. Not gone through the readme files yet. Will do that when the config is sorted.

**Pull request closed without merging**